### PR TITLE
Add overlay zookeeper test

### PR DESCRIPTION
**Pull request by chenchun**
Signed-off-by: Chun Chen <ramichen@tencent.com>

Depends on https://github.com/docker/libkv/pull/82

### CODE REVIEW

File: test/integration/dnet/helpers.bash [hunk 1]
```diff
@@ -194,3 +215,61 @@ function runc() {
...
     dnet_exec ${dnet} "umount /var/run/netns/c && rm /var/run/netns/c"
 }
+
+function start_zookeeper() {
+    stop_zookeeper
+    docker run -d \
+	   --name=zookeeper_server \
+	   -p 2181:2181 \
+	   -h zookeeper \
+	   jplock/zookeeper
```
Comment (abronan) [inline test/integration/dnet/helpers.bash]: Can you use `dnephin/docker-zookeeper:3.4.6` instead? Just to make sure we are aligned with other projects and we test from the same source ;)

**Comment by mrjana** [comment]
@chenchun Thanks for the PR. Much appreciated.

Looks like overlay tests are not getting skipped in circleci. Since circleci is running a very old linux kernel it can't run overlay tests and trying to invoke skip_for_circleci inside helpers.bash does not work. Can you try adding skip_for_cicleci in each overlay test in overlay.bats and zookeeper.bats?

**Comment by mrjana** [comment]
@chenchun Can you also update the Godep to update libkv(if there is any zookeepr fix) so that this test works for everybody

**Comment by chenchun** [comment]
@mrjana Addressed your comments and changed host port of zk container to 2182 as I found 2181 is used by other process on CI.

File: test/integration/dnet/helpers.bash (hunk 1 above)
Comment (chenchun) [inline test/integration/dnet/helpers.bash]: yes, changed.

File: test/integration/dnet/overlay.bats [hunk 2]
```diff
@@ -3,40 +3,9 @@
...
-    done
-
-    # Teardown the container connections and the network
-    for i in `seq ${start} ${end}`;
-    do
-	net_disconnect ${i} container_${i} multihost
-	dnet_cmd $(inst_id2port $i) container rm container_${i}
-    done
-
+    run test_overlay overlay
```
Comment (mavenugo) [inline test/integration/dnet/overlay.bats]: did you mean to pass `consul` as the param instead of `overlay` ?

Comment (chenchun) [inline test/integration/dnet/overlay.bats]: ok, changed it to `consul`

**Comment by mrjana** [comment]
@chenchun I rebased and merged your commit manually. Thanks fo the PR. Closing this.

**Pull request closed without merging**